### PR TITLE
adding Androguard analyzer closes (#2514)

### PR DESCRIPTION
**Pull request by spoiicy**
# Description

Adding Androguard analyzer which is a python tool to reverse engineer APKs.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [ x] I have read and understood the rules about [how to Contribute](https://intelowlproject.github.io/docs/IntelOwl/contribute/) to this project
- [ x] The pull request is for the branch `develop`
- [ x] A new plugin (analyzer, connector, visualizer, playbook, pivot or ingestor) was added or changed, in which case:
    - [ x] I strictly followed the documentation ["How to create a Plugin"](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-add-a-new-plugin)
    - [ ] [Usage](https://github.com/intelowlproject/docs/blob/main/docs/IntelOwl/usage.md) file was updated.
    - [ ] [Advanced-Usage](https://github.com/intelowlproject/docs/blob/main/docs/IntelOwl/advanced_usage.md) was updated (in case the plugin provides additional optional configuration).
    - [ x] I have dumped the configuration from Django Admin using the `dumpplugin` command and added it in the project as a data migration. (["How to share a plugin with the community"](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-share-your-plugin-with-the-community))
    - [ ] If a File analyzer was added and it supports a mimetype which is not already supported, you added a sample of that type inside the archive `test_files.zip` and you added the default tests for that mimetype in [test_classes.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/api_app/analyzers_manager/test_classes.py).
    - [ x] If you created a new analyzer and it is free (does not require any API key), please add it in the `FREE_TO_USE_ANALYZERS` playbook by following [this guide](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-modify-a-plugin).
    - [ ] Check if it could make sense to add that analyzer/connector to other [freely available playbooks](https://intelowlproject.github.io/docs/IntelOwl/usage/#list-of-pre-built-playbooks).
    - [ x] I have provided the resulting raw JSON of a finished analysis and a screenshot of the results.
    - [ ] If the plugin interacts with an external service, I have created an attribute called precisely `url` that contains this information. This is required for Health Checks. 
    - [ ] If the plugin requires mocked testing, `_monkeypatch()` was used in its class to apply the necessary decorators.
    - [ ] I have added that raw JSON sample to the `MockUpResponse` of the `_monkeypatch()` method. This serves us to provide a valid sample for testing.
- [ ] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [ x] Linters (`Black`, `Flake`, `Isort`) gave 0 errors. If you have correctly installed [pre-commit](https://intelowlproject.github.io/docs/IntelOwl/contribute/#how-to-start-setup-project-and-development-instance), it does these checks and adjustments on your behalf.
- [ ] I have added tests for the feature/bug I solved (see `tests` folder). All the tests (new and old ones) gave 0 errors.
- [ ] If changes were made to an existing model/serializer/view, the docs were updated and regenerated (check [CONTRIBUTE.md](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Contribute.md)).
- [ ] If the GUI has been modified:
    - [ ] I have a provided a screenshot of the result in the PR.
    - [ ] I have created new frontend tests for the new component or updated existing ones.
- [ ] After you had submitted the PR, if `DeepSource`, `Django Doctors` or other third-party linters have triggered any alerts during the CI checks, I have solved those alerts.

### Important Rules
- If you miss to compile the Checklist properly, your PR won't be reviewed by the maintainers.
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review. After being reviewed and received a "change request", you should explicitly ask for a review again once you have made the requested changes.

### Screenshot
Screenshot 2024-10-28 at 11 50 03 AM

### Full Report
[job#6_report (1).json](https://github.com/user-attachments/files/17539058/job.6_report.1.json)

### CODE REVIEW

File: api_app/analyzers_manager/migrations/0124_analyzer_config_androguard.py
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0124_analyzer_config_androguard.py:27]: you can also add the following ones:
application/x-dex
application/zip
application/java-archive

That's because sometimes they get flagged like that. Also other APK tools use this list

File: api_app/analyzers_manager/migrations/0124_analyzer_config_androguard.py
Comment (mlodic) [inline api_app/analyzers_manager/migrations/0124_analyzer_config_androguard.py:25]: this can be RED cause this tool is internal so there's no risk of information leakage

**Review by mlodic — CHANGES_REQUESTED** [review]
cool! I'll add some other useful functions:
* get_receivers()
* get_services()
* is_valid_APK()
* get_min_sdk_version()
* get_max_sdk_version()
* get_target_sdk_version()
* get_androidversion_code()
* get_androidversion_name()
In that way we can have a more broader overview of the analyzed file

File: requirements/project-requirements.txt
Comment (spoiicy) [reply requirements/project-requirements.txt:89]: Using version >=4.x will result in dependency conflict as follows:
` INFO: pip is looking at multiple versions of quark-engine to determine which version is compatible with other requirements. This could take a while.`
`30.49 ERROR: Cannot install -r project-requirements.txt (line 70) and androguard==4.1.2 because these package versions have conflicting dependencies.`
`30.49 The conflict is caused by:`
`30.49     The user requested androguard==4.1.2 and quark-engine 24.10.1 depends on androguard==3.4.0a1`
`30.49 To fix this you could try to:`
 `1. loosen the range of package versions you've specified`
 `2. remove package versions to allow pip to attempt to solve the dependency conflict`

All the functions that you've mentioned can be implemented in the similar way in version `3.4.0a1` as well

**Review by spoiicy — COMMENTED** [review]
The additional functionalities mentioned by @mlodic have been implemented along with modifications to TLP value and accepted file mime-types.

File: requirements/project-requirements.txt
Comment (mlodic) [reply requirements/project-requirements.txt:89]: thanks for the explanation.

Could you please add this information as a comment here? That would help the maintainers for the future

**Review by mlodic — CHANGES_REQUESTED** [review]
please open a PR for the docs alongside this one then I guess we are done

**Comment by spoiicy** [comment]
I've opened a PR (https://github.com/intelowlproject/docs/pull/9) for the addition of Androguard usage details.

**Comment by mlodic** [comment]
just the last thing requested from deepsource: you should create the `update` method with a pass to respect the abstract method

**Comment by spoiicy** [comment]
@mlodic I've added the **update** method with a pass. Hopefully, this will resolve the issue and we can merge this PR :)

**Comment by mlodic** [comment]
there's left an unrelated error in the tests. I'll check it before merging

**Pull request merged**
